src/heap.c

```diff
@@ -31,13 +31,40 @@ static void *defaultRealloc(void *data, void *ptr, size_t size)
 static void *defaultAlignedAlloc(void *data, size_t alignment, size_t size)
 {
     (void)data;
+#ifdef _WIN32
+    /*
+        Windows does not have support for aligned_alloc, the closest functionality
+        is provided by _aligned_malloc(), but the returned memory has to be freed with
+        _aligned_free(). Freeing this with free() would not properly clean up the memory.
+        TODO: This should be addressed
+    */
+    return _aligned_malloc(size, alignment);
+#elif defined(__APPLE__)
+    void * p1; // original block
+	void ** p2; // aligned block
+	size_t offset = alignment + sizeof(void *) - 1;
+	if ((p1 = (void *)malloc(size + offset)) == NULL)
+		return NULL;
+	p2 = (void **)(((size_t)(p1) + offset) & ~(alignment - 1));
+	p2[-1] = p1;
```

Comment:
What does this line do?

Reply:
Honestly I don't know - aligned malloc is way beyond my C knowledge) But it's the best answer I found: https://stackoverflow.com/a/38089003 : The line stores the original pointer before the aligned one, if I did understand it correctly.

Reply:
Hehe, yeah, it looks a bit fishy at first sight but it's only pointer arithmetic, so is indeed fine.
There's a comment in the stackoverflow thread 
```
Also: size_t (in the line that sets p2) should be uintptr_t. There's no guarantee that size_t is large enough to represent pointer values.
```
Could you add that please?

Reply:
Oh, good eye! Thank you - just fixed it.